sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -345,6 +346,46 @@ class DataFrameReader private[sql](sqlContext: SQLContext) extends Logging {
   }
 
   /**
+   * Loads a CSV file and returns the result as a [[DataFrame]].
+   *
+   * This function goes through the input once to determine the input schema. If you know the
+   * schema in advance, use the version that specifies the schema to avoid the extra scan.
+   *
+   * @since 2.0.0
+   */
+  def csv(paths: String*): DataFrame = format("csv").load(paths : _*)
```

Comment:
I'd keep only the path one, and remove the rest.

Also you need to add the scala annotation to enable calling this in Java.